synfig-studio/src/gui/docks/dock_timetrack2.cpp

```diff
@@ -233,6 +233,74 @@ void Dock_Timetrack2::setup_tool_palette()
 		action_button_map[tool_button->get_name()] = tool_button;
 		tool_item_group->add(*tool_button);
 	}
+    Gtk::RadioButtonGroup interpolation_group;
+    //button1
+	Gtk::ToolButton *clamped_button = manage(new Gtk::ToolButton(
+													*manage(new Gtk::Image(
+																Gtk::StockID("synfig-interpolation_type_clamped"),

```

Comment:
Let's avoid deprecated `StockID`.
create default `Image`; and right after call its `set_from_icon_name()` method. Don't use `Image(icon_name)` constructor because it is only available after gtkmm 3.24, and we intend to support 3.18.

Reply:
for some reason doing that the icon gives the image it gives when the icon isn't found, I went to check how the interpolation icons are set. it seems they are not set the regular way I know of, they are all in the same .sif and there seems to be some configurations done to them.

Reply:
The problem probably was the icon name you were using.
These are the real icon names:
https://github.com/synfig/synfig/blob/ac8214e47fd71dbec1066f475e98bd9dda5859e7/synfig-studio/src/gui/iconcontroller.cpp#L335-L357

Please remove the StockID dependency.

Reply:
yes these were the same names I tried it still gives out the image missing icon for some reason :(

Reply:
I guess it is that way because your PR is based on a commit previous 865f1977d0a3e4129a1031cc39d7ea7514bc7158 :)